asf_public_discourse_home_decarbonisation/analysis/mse/initial_text_analysis_category_data.py

```diff
@@ -0,0 +1,447 @@
+"""
+This script performs an initial text analysis on Money Saving Expert data, by:
+- Computing the frenquency os specific words and n-grams of interest;
+- Identiyfying top words and n-grams;
+
+Plots are saved in the folder `asf_public_discourse_home_decarbonisation/outputs/figures/mse/`.
+
+To run this script:
+`python asf_public_discourse_home_decarbonisation/analysis/mse/initial_text_analysis_category_data.py`
+
+To change category or date/time range, use the following arguments:
+`python asf_public_discourse_home_decarbonisation/analysis/mse/initial_text_analysis_category_data.py --category <category> --collection_date_time <collection_date_time>`
+where
+<category>: category or sub-forum to be analysed, e.g. "energy", "lpg-heating-oil-solid-other-fuels", etc.
+<collection_date_time>: data collection date/time in the format YYYY_MM_DD
+"""
+
+# ## Package imports
+import warnings
+
+# Ignoring warnings from Pandas
+warnings.simplefilter(action="ignore")
+
+import numpy as np
+import pandas as pd
+from nltk.tokenize import word_tokenize
+import os
+import asf_public_discourse_home_decarbonisation.utils.plotting_utils as pu
+import asf_public_discourse_home_decarbonisation.config.plotting_configs as pc
+from asf_public_discourse_home_decarbonisation.getters.mse_getters import get_mse_data
+from asf_public_discourse_home_decarbonisation.utils.text_processing_utils import (
+    process_text,
+    create_ngram_frequencies,
+    english_stopwords_definition,
+    lemmatize_sentence,
+    frequency_ngrams,
+)
+from asf_public_discourse_home_decarbonisation.utils.plotting_utils import (
+    plot_and_save_top_ngrams,
+    plot_and_save_wordcloud,
+)
+from asf_public_discourse_home_decarbonisation import PROJECT_DIR
+
+import argparse
+import logging
+
+logger = logging.getLogger(__name__)
+
+# Analysis-specific parameters
+number_ngrams_wordcloud = 30
+min_frequency_tokens = 100
+min_frequency_bigrams = 50
+min_frequency_trigrams = 10
+top_ngrams_barplot = 10
+
+# Specific groups of keywords we want to look at
+keyword_dictionary = {
```

Comment:
This is obviously very subjective. It all started with the exploration work we did in the beginning of the year, where we collected data from Twitter. When you collect data from Twitter, you need to define keywords of interest. I talked to a few members of the mission team and we came up with a good set for data collection. That set was the starting point for this :)

Anything important missing?

Reply:
Sounds like a good place to start.

When I have done similar analysis in the past - I took a sample of the data and manually read through to get an idea of what might appear as keywords. This is a good way to identify variations you might not think of. You could do this by sampling threads discussing topics in each of your categories of interest.

A few that might be missing, just some suggestions, including some that came up from looking at 2 random HP threads on MSE:

- hp
- hps
- ASHeat pump
- air to air
- a2a
- heatpump
- h2 (for hydrogen)
- subsidy
- Perhaps 'contractor' can appear in the 'installer_keywords' category although this might be too generic
- gas central heating/ gas CH/GCH/CH
- expensive/cheap - although this can capture a lot of irrelevant stuff

Reply:
Just to give a bit of context @crispy-wonton: This was never supposed to be a very robust analysis. The idea was to get an initial sense of themes in each specific sub-forum. We then did the analysis of top tokens and n-grams and used BERTopic to identify themes (will follow up in a next PR). I added a few of the ones you added above, thank you. Not all though, as at this point i'd rather have more false negatives than false positives. :)